### DISCUSSION TITLE
Rspack version of react with-module-federation

### DISCUSSION THREAD

**Discussion by Elvis116** [nrwl/nx#22615] [Feature Requests]
Need Rspack version of react with-module-federation

**Comment by Coly010** [comment]
There are internal plans already regarding this feature.

We'll keep everyone updated with progress

**Comment by soumyanildas** [comment]
I have done a setup with Rspack and Module Federation 2.0 but I have remotes as separate repos as that was our requirement.

Maybe you could try the same in a monorepo setup?

Check my blog here - https://blog.soumyanildas.com/nx-rspack-micro-frontend

But there's an issue with sass module only working on 18.0.6 @nx/rspack

**Comment by Wykks** [comment]
Look like it's done: https://github.com/nrwl/nx/pull/27696

But if it's as bad as the support with webpack (try secondary entry point with module federation with something else than Angular), I'd rather do it myself :/

**Reply by Coly010** [reply]
If you have issues with secondary entry points with react + MF, raise an issue and tag me.

**Reply by Wykks** [reply]
Already did: https://github.com/nrwl/nx/issues/26688#issuecomment-2254205548

Put maybe I should be more specific, because this issue is about the lint rule, but the issue is beyond the rule issue. But you closed the issue a bit too fast :sweat_smile:

**Comment by Coly010** [comment]
This has been added in Nx 19.7.0
